packages/go_router/lib/src/router.dart

```diff
@@ -133,13 +134,32 @@ class GoRouter extends ChangeNotifier with NavigatorObserver {
 
   /// Navigate to a URI location w/ optional query parameters, e.g.
   /// `/family/f2/person/p1?color=blue`
-  void go(String location, {Object? extra}) {
+  void go(
+    String location, {
+    Map<String, String> params = const <String, String>{},
+    Map<String, String> queryParams = const <String, String>{},
+    Object? extra,
+  }) {
     assert(() {
-      log.info('going to $location');
+      log.info(
+        'going to "$location"'
+        '${params.isEmpty ? '' : ', params: $params'}'
+        '${queryParams.isEmpty ? '' : ', queryParams: $queryParams'}',
+      );
       return true;
     }());
+    late final String parsedLocation;
+    if (params.isNotEmpty || queryParams.isNotEmpty) {
+      parsedLocation = parsePathWithParameters(
+        location,
+        params: params,
+        queryParams: queryParams,
+      );
+    } else {
+      parsedLocation = location;
+    }
     _routeInformationProvider.value =
```

Comment:
I had to do this to not make it a breaking change.

If it goes through `parsePathWithParameters` all the time and `location = '/family?fid=f2'` (for example), the location given to `RouteInformation` will be `'/family%3Ffid=f2'`